iterative_assembly.wdl

```diff
@@ -0,0 +1,110 @@
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import "tasks/bwa.wdl" as bwa
+import "tasks/samtools.wdl" as samtools
+import "tasks/spades.wdl" as spades
+import "tasks/biopet.wdl" as biopet
+import "tasks/seqtk.wdl" as seqtk
+import "tasks/picard.wdl" as picard
+
+# This workflow takes an existing assembly and reads that have already passed QC.
+# It maps the reads back to the assembly and uses the mapped reads to create a new assembly.
+# This workflow can be run on its own spades.contigs or spades.scaffolds output
+workflow iterativeAssembly {
+    File inputAssembly
+    File read1
+    File? read2
+    Int? subsampleNumber
+    Int? subsampleSeed
+    String outputDir
+
+    # First index the assembly
+    call bwa.index as bwaIndex {
+        input:
+            fasta = inputAssembly,
+            outputDir = outputDir + "/bwaIndex"
+    }
+
+    # Map the reads back to the assembly.
+    call bwa.mem as bwaMem {
+        input:
+            inputR1 = read1,
+            inputR2 = read2,
+            referenceFasta = bwaIndex.indexedFasta,
+            indexFiles = bwaIndex.indexFiles,
+            outputPath= outputDir + "/" + "ReadsMappedToInputAssembly.bam"
+    }
+
+    # Get the reads that mapped to the assembly. This means filtering out the UNMAP flag.
+    # QCFAIL is also filtered out. For paired reads, the PROPER_PAIR flag is used.
+    call samtools.view as selectMappedReads {
+        input:
+            inFile= bwaMem.bamFile,
+            outputBam = true,
+            excludeSpecificFilter = 12, # UNMAP,MUNMAP should both be present for the read to be filtered out.
+            outputFileName = outputDir + "/unmapppedReadsFiltered.bam"
+    }
+
+    call picard.SamToFastq as SamToFastq {
+        input:
+            inputBam = selectMappedReads.outputFile,
+            outputRead1 = outputDir + "/filtered_reads1.fq.gz",
+            outputRead2 = if defined(read2) then outputDir + "/filtered_reads2.fq.gz" else read2
+    }
+
+    # Allow subsampling in case number of mapped reads is too much for the assembler to handle.
+    if (defined(subsampleNumber)) {
+        call seqtk.sample as subsampleReads1 {
+            input:
+                sequenceFile = SamToFastq.read1,
+                number = subsampleNumber,
+                seed = subsampleSeed,
+                outFilePath = outputDir + "/subsampling/subsampledReads1.fq.gz",
+                zip=true
```

Comment:
should zip not be compress? this what we use in other tasks also

Reply:
We can change this tasks if necessary. The default is true, and should not really be changed